etl/steps/data/garden/un/2023-10-30/un_members.py

```diff
@@ -0,0 +1,217 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import owid.catalog.processing as pr
+from owid.catalog import Table
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Define latest year for the dataset (the year of the last 31 December).
+LATEST_YEAR = 2022
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("un_members")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["un_members"].reset_index()
+
+    #
+    # Process data.
+    tb = data_processing(tb)
+
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Add missing countries
+    tb = add_missing_countries(tb)
+
+    # Add regional aggregations
+    tb = regional_aggregations(tb)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def data_processing(tb: Table) -> Table:
+    """
+    Create membership_status column from admission year.
+    """
+
+    # Create a year column by creating a range between the year of admission and LATEST_YEAR.
+    tb["year"] = tb["admission"].astype(int).apply(lambda x: range(tb["admission"].min(), LATEST_YEAR + 1))
+
+    # Explode the year column.
+    tb = tb.explode("year")
+
+    # Create membership_status column, which is "Member" when year is greater or equal to admission year, and "Not a member" otherwise.
+    # I copy the admission column first to keep metadata
+
+    tb["membership_status"] = tb["admission"].copy()
+    tb.loc[tb["year"] < tb["admission"], "membership_status"] = "Not a member"
+    tb.loc[tb["year"] >= tb["admission"], "membership_status"] = "Member"
+
+    # Drop admission column.
+    tb = tb.drop(columns=["admission"])
+
+    return tb
+
+
+def add_missing_countries(tb: Table) -> Table:
+    """
+    Add countries not available in the list of UN members
+    """
+
+    # Load region dataset
+    tb_regions = paths.load_dependency("regions", channel="grapher")
+    tb_regions = tb_regions["regions"].reset_index()
+    tb_regions = tb_regions[
```

Comment:
There;s this error here, @Marigold:
```
/Users/parriagadap/etl/etl/steps/data/garden/un/2023-10-30/un_members.py
  /Users/parriagadap/etl/etl/steps/data/garden/un/2023-10-30/un_members.py:86:18 - error: "__getitem__" method not defined on type "Snapshot" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 8.308sec
make: *** [check-typing] Error 1
```

Should I just add `# type: ignore`? I needed to use load_dependency, because I am calling both garden and grapher steps for regions.

Reply:
Correct, you shouldn't be ashamed of using `# type: ignore`. I always do it when I'm lazy to fix type hints...